source/component/PasDoc_Parser.pas

```diff
@@ -1336,8 +1337,20 @@ procedure TParser.ParseTVCSection(U: TPasUnit);
                 end;
               pmType:
                 begin
-                  Scanner.UnGetToken(t);
-                  ParseType(U);
+                  //In latest FPC version it is posible to define generic functions and procedures
+                  if t.IsStandardDirective(SD_GENERIC) then
+                  begin
+                    t2 := PeekNextToken;
+                    if t2.IsKeyWord(KEY_FUNCTION) or t2.IsKeyWord(KEY_PROCEDURE) then
+                      Break
+                    else
+                      ParseType(U, t.Data)
+                  end
+                  else
+                  begin
+                    Scanner.UnGetToken(t);
+                    ParseType(U, EmptyStr)
```

Comment:
In PasDoc code we just use literal `''`, we don't use `EmptyStr`. I would prefer we keep using `''`, it seems simpler (although I do understand that `EmptyStr` may look more explicit, so less potential for errors).

Also FPC 3.0.4 doesn't define `EmptyStr` (though this could be easily fixed).

Please change it to simple `''`.